server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -81,6 +102,102 @@ func (b *BuildEventHandler) OpenChannel(ctx context.Context, iid string) interfa
 		hasReceivedStartedEvent: false,
 		eventsBeforeStarted:     make([]*inpb.InvocationEvent, 0),
 		logWriter:               logWriter,
+		onClose:                 onClose,
+	}
+}
+
+func (b *BuildEventHandler) GetStatsRecorder() *statsRecorder {
+	return b.statsRecorder
+}
+
+// recordStatsRequest is a request to record the stats for an invocation, which
+// will be fulfilled by one of the workers spawned by StatsRecorder.
+type recordStatsRequest struct {
+	invocationID string
+	requestedAt  time.Time
```

Comment:
nit: can you add a comment clarifying what requestedAt means?

Reply:
Done (and renamed to createdAt)